plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSynchronizerModule.java

```diff
@@ -41,8 +43,11 @@ public void configure(Binder binder)
 
         // S3
         jsonCodecBinder(binder).bindJsonCodec(S3NativeTransactionLogSynchronizer.LockFileContents.class);
-        synchronizerBinder.addBinding("s3").to(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
-        synchronizerBinder.addBinding("s3a").to(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
-        synchronizerBinder.addBinding("s3n").to(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
+        binder.bind(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
+        binder.bind(S3ConditionalWriteLogSynchronizer.class).in(Scopes.SINGLETON);
+
+        synchronizerBinder.addBinding("s3").to(S3SwitchingTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
```

Comment:
why do we need `S3SwitchingTransactionLogSynchronizer`? We should be able to bind either Native or ConditionalWriter synchronizer based on what used S3 supports